File: test/cases/record/test_record_statistics.pas
program test;
type
  stats = record
    count: integer;
    sum: real;
    avg: real
  end;
var
  s: stats;
  i: integer;

begin
  s.count := 0;
  s.sum := 0.0;

  for i := 1 to 5 do
  begin
    s.count := s.count + 1;
    s.sum := s.sum + i
  end;

  s.avg := s.sum / s.count;

  write(s.count);
  write(s.sum);
  write(s.avg)
end.

File: test/cases/record/test_record_statistics.c
#include <stdio.h>
typedef struct {
    int count;
    float sum;
    float avg;
} stats;
stats s;

int i;


int main(void) {
    s.count = 0;
    s.sum = 0.0;
    for (i = 1; i <= 5; i++) {
        s.count = (s.count + 1);
        s.sum = (s.sum + i);
    }
    s.avg = (s.sum / s.count);
    printf("%d", s.count);
    printf("%f", s.sum);
    printf("%f", s.avg);

    return 0;
}
